libs-scala/resources-akka/src/main/scala/com/digitalasset/resources/akka/CancellableResourceOwner.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.resources.akka
+
+import akka.actor.Cancellable
+import com.daml.resources.{AbstractResourceOwner, HasExecutionContext, ReleasableResource, Resource}
+
+import scala.concurrent.Future
+
+class CancellableResourceOwner[Context: HasExecutionContext](acquireCancellable: () => Cancellable)
+    extends AbstractResourceOwner[Context, Cancellable] {
+  override def acquire()(implicit context: Context): Resource[Context, Cancellable] =
+    ReleasableResource(Future(acquireCancellable()))(cancellable =>
```

Comment:
I understand it's convenient here but I feel like exposing a cancellation as a resource release changes the connotation (and thus hinders readability), as a cancellation can (and perhaps usually will) also trigger some kind of failure. Is this change much worse off without this additional owner?

Reply:
I wasn't aware that cancellations can trigger failures. Typically resources are only released when the application is shutting down; isn't this the kind of occasion where if there is something pending, it _should_ get some kind of cancellation error?

Reply:
Akka doesn't seem to be explicit (for a change, although I must admit that I haven't looked around too long) about whether a `Cancellable` can throw; I was more thinking about (Java) thread interruption though. OK to leave it for me.